fasthttputil/inmemory_listener.go

```diff
@@ -60,12 +66,23 @@ func (ln *InmemoryListener) Close() error {
 	return err
 }
 
+type inmemoryAddr int
+
+func (inmemoryAddr) Network() string {
+	return "inmemory"
+}
+
+func (inmemoryAddr) String() string {
+	return "InmemoryListener"
+}
+
 // Addr implements net.Listener's Addr.
 func (ln *InmemoryListener) Addr() net.Addr {
-	return &net.UnixAddr{
-		Name: "InmemoryListener",
-		Net:  "memory",
+	if ln.listenerAddr != nil {
+		return ln.listenerAddr
```

Comment:
Do `SetLocalAddr()` and `Addr()` need to be considered for concurrency?

Reply:
You could argue that it needs to be. However, I thought the only meaningful way to set the address would be to do it before the listener is used in any way. I just did not add it to `NewInmemoryListener()` to leave this completely optional.

It is probably better to guard against data races, though and do not leave this to the caller. I will implement it, thanks!